### ISSUE TITLE
E0001 when LayerVersion ContentUri references local file

### ISSUE THREAD

**Issue by bryanburke** [aws-cloudformation/cfn-python-lint#755]
*cfn-lint version: 0.16.0*

Using latest version of cfn-lint and updated spec files.

*Description of issue.*

Linting a SAM template fails when including a LayerVersion with `ContentUri` referencing a local file.

Example template:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Transform: AWS::Serverless-2016-10-31
Resources:
    ExampleLayer:
        Type: AWS::Serverless::LayerVersion
        Properties:
            LayerName: "Example"
            ContentUri: "./layers/example.zip"
            CompatibleRuntimes:
                - python3.6
```

Error:

```
E0001 Error transforming template: Resource with id [ExampleLayere5b6fe5ccd] is invalid. 'ContentUri' is not a valid S3 Uri of the form "s3://bucket/key" with optional versionId query parameter.
example.yaml:1:1
```

However, this template validates with `sam validate` and works as expected with `sam package`. Example output template from `sam package`:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  ExampleLayer:
    Properties:
      CompatibleRuntimes:
      - python3.6
      ContentUri: s3://<redacted>/ca1212f406e4a73dbe62c2ac6b8b02f0
      LayerName: Example
    Type: AWS::Serverless::LayerVersion
Transform: AWS::Serverless-2016-10-31
```

I will note that even though this is the expected `sam package` behavior, I do not believe it is properly documented in either SAM CLI or AWS CLI (`aws cloudformation package`) docs.
